### PR TITLE
Add FieldsContainer.animate()

### LINKED ISSUE #405: Add animations to Core

### PR DESCRIPTION
**Pull request by PProfizi**
Resolves #405 
- [x] Implement helper
- [x] Tests
- [x] Example
- [x] GIFs are included in the HTML doc correctly

### CODE REVIEW

**Review by cbellot000 — COMMENTED** [review]
Thanks for this enhancement, if you could maybe rename the PR to something generic?
Also maybe you've planned to fix it, but code coverage is complaining a lot :/

File: ansys/dpf/core/animator.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+class _InternalAnimatorFactory:
+    """
+    Factory for _InternalAnimator based on the backend."""
+    @staticmethod
+    def get_animator_class():
+        return _PyVistaAnimator
+
+
+class _PyVistaAnimator(_PyVistaPlotter):
```
[the remark this answers is not in the document]
Comment (PProfizi) [reply ansys/dpf/core/animator.py:23]: Indeed, I think I forgot to remove this. At the start, I wanted to reuse some methods of the _PyVistaPlotter.

File: ansys/dpf/core/fields_container.py [hunk 2]
```diff
@@ -488,6 +489,114 @@ def get_time_scoping(self):
...
+            Defaults to None, which takes self if possible. Set as False to force static animation.
+        scale_factor : float, list, optional
+            Scale factor to apply when warping the mesh. Defaults to 1.0. Can be a list to make
+            scaling frequency-dependent.
+        through: string, optional
+            Label to iterate through to create the animation. Defaults to "time".
+        """
+        from ansys.dpf.core.animator import Animator
+        # Take the norm of the fields if vector
+        if self[0].component_count > 1:
```
[the remark this answers is not in the document]
Comment (PProfizi) [reply ansys/dpf/core/fields_container.py:514]:
```
# Merge field data into a single array
if component_count > 1:
    overall_data = np.full((len(mesh_location), component_count), np.nan)
else:
    overall_data = np.full(len(mesh_location), np.nan)

for field in fields_container:
    ind, mask = mesh_location.map_scoping(field.scoping)
    overall_data[ind] = field.data[mask]
```
@cbellot000 Is that what you were thinking of?

File: ansys/dpf/core/animator.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+            Name of the workflow output to use as Field for each frame's contour.
+            Default to "to_render".
+        save_as : str, optional
+            Path of file to save the animation to. Defaults to None. Can be of any format supported
+            by pyvista.Plotter.write_frame (.gif, .mp4, ...).
+        scale_factor : float, list, optional
+            Scale factor to apply when warping the mesh. Defaults to 1.0. Can be a list to make
+            scaling frequency-dependent.
+        **kwargs : optional
+            Additional keyword arguments for the animator.
```
[the remark this answers is not in the document]
Comment (PProfizi) [reply ansys/dpf/core/animator.py:217]: Improved the docstrings. Mainly they are PyVista kwargs.

File: ansys/dpf/core/animator.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+        """
+        _InternalAnimatorClass = _InternalAnimatorFactory.get_animator_class()
+        self._internal_animator = _InternalAnimatorClass(**kwargs)
+        self._workflow = workflow
+
+    @property
+    def workflow(self) -> core.Workflow:
+        """
+        Workflow used by the Animator to generate a Field for each frame.
```
[the remark this answers is not in the document]
Comment (PProfizi) [reply ansys/dpf/core/animator.py]: Indeed, corrected.

**Comment by PProfizi** [comment]
> Thanks for this enhancement, if you could maybe rename the PR to something generic? Also maybe you've planned to fix it, but code coverage is complaining a lot :/

Will be improving the code coverage, I also think we should make the coverage criterion more strict or per file. I did not realize one of my files was covered 70%... this is low.
@cbellot000 concerning the PR's name, I wanted to specify the main feature added, but I can mention the Animator.

**Review by cbellot000 — APPROVED** [review]

**Pull request merged**